src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.cs

```diff
@@ -266,6 +266,12 @@ await TrySetDigestAuthToken(request, challenge.Credential, digestResponse, isPro
                         break;
 
                     case AuthenticationType.Basic:
+                        if (CredentialCache.DefaultCredentials == credentials)
```

Comment:
Should we duplicate this for AuthenticationType.Digest as well? It would have same problem, right?